proposals/0016-simd-application-fees.md

```diff
@@ -0,0 +1,149 @@
+---
+simd: '0016'
+title: Application Fees (Write-lock fees)
+authors:
+  - Godmode Galactus (Mango Markets)
+  - Maximilian Schneider (Mango Markets)
+category: Fees
+type: Fees
+status: Draft
+created: 2022-12-23
+---
+
+## Problem
+
+According to the discussion on the following issue:
+https://github.com/solana-labs/solana/issues/21883
+
+During network congestion, many transactions in the network cannot be processed 
+because they all want to write lock similar accounts. 
+When a write lock on an account is taken by a transaction batch no other batch 
+can use this account in parallel, so only transactions belonging to a single batch 
+are processed correctly and all others are forwarded to the leader of the next slot. 
+There are multiple accounts of OpenBook (formerly serum), mango markets, etc which 
+are used in high-frequency trading. During the event of extreme congestion we observe 
+that specialized trading programs write lock these accounts but never CPI into the respecitve 
+programs unless they can extract profit, effectively starving the actual users for access. 
```

Comment:
> `effectively starving the actual users for access.`

I'm just curious. I think users can just out-bid these specialized trading program's transactions with priority fees?

Reply:
It again comes to batching of transactions i guess. If there are lot of spamming transactions in the cluster even with the priority fees there is high chance the write lock has already been taken. But yeah i will interesting to see how priority fees affect life cycle of transaction.